spark/v3.0/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/AssignmentAlignmentSupport.scala

```diff
@@ -100,7 +99,7 @@ trait AssignmentAlignmentSupport {
             case StructType(fields) =>
               // build field expressions
               val fieldExprs = fields.zipWithIndex.map { case (field, ordinal) =>
-                createAlias(GetStructField(col, ordinal, Some(field.name)), field.name)
+                Alias(GetStructField(col, ordinal, Some(field.name)), field.name)()
```

Comment:
I think that we should leave the reflection in 3.0. We aren't renaming the `iceberg-spark3-runtime` Jar so it is possible that people currently using it for 3.1 will continue to. As long as we can maintain compatibility with that by doing nothing, I think we should. I don't think that it is necessary to do much else, but I don't see a reason to remove the compatibility.

Reply:
Ok, I reverted the code changes for spark/v3.0. I kept the change to spark/v3.0/build.gradle as that is just cleanup.

Reply:
Thanks!